lib/Dialect/Calyx/CalyxOps.cpp

```diff
@@ -1268,6 +1270,253 @@ SmallVector<DictionaryAttr> InstanceOp::portAttributes() {
   return portAttributes;
 }
 
+//===----------------------------------------------------------------------===//
+// PrimitiveOp
+//===----------------------------------------------------------------------===//
+
+/// Lookup the component for the symbol. This returns null on
+/// invalid IR.
+hw::HWModuleExternOp PrimitiveOp::getReferencedPrimitive() {
```

Comment:
Would returning `hw::HWModuleLike` work? I think that can be generalized pretty easily as a follow up, just curious if it could work off the bat. This way, the emitter would depend on an interface, not an implementation. The interface is relatively new, so if this is a pain today, no worries.

Reply:
HWModuleLike doesn't seem to have any way to access it's ports, so I don't think it's usable for now.

Reply:
Sounds good. I guess some of those ended up in HWMutableModuleLike, but we should just make sure HWModuleLike has what it needs here before doing this generalization. Thanks for taking a look.